src/classes/CRLP_RollupQueryBuilder.cls

```diff
@@ -0,0 +1,312 @@
+            }
+            if (!String.isEmpty(addlFilter)) {
+                whereClauses.add(addlFilter);
+            }
+
+            if (rollupType != CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
+                if (householdSettings.npo02__Always_Rollup_to_Primary_Contact__c == false) {
+                    whereClauses.add('(IsPrimary = False OR Opportunity.Account.npe01__SYSTEMIsIndividual__c = False)');
+                } else {
+                    whereClauses.add('IsPrimary = False');
+                }
+            }
+
+            orderByFields.add(keyFieldOverrideForQuery);
+            if (rollupType != CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
+                orderByFields.add('Opportunity.CreatedDate ASC');
+            } else {
+                orderByFields.add('Opportunity__r.CreatedDate ASC ');
+            }
+
+        } else {
+            // Hard Credit Rollups
+            query = buildFullDetailsQueryForSummaryRecords();
+
+            if (query.containsIgnoreCase('ORDER BY ')) {
+                orderByFields.add(query.split('ORDER BY')[1]);
+                query = query.split('ORDER BY')[0];
+            }
+
+            if (query.containsIgnoreCase(' WHERE ')) {
+                whereClauses.add(query.split(' WHERE ')[1]);
+                query = query.split(' WHERE ')[0];
+            }
+
+            if (rollupType != CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
+                // Identify a common filter that can be applied to the main query to reduce the number of records queried
+                String addlFilter = CRLP_Query_SEL.buildCommonQueryFilterOnOpportunity(summaryObjectType,
+                        Opportunity.SObjectType, CRLP_Rollup_SEL.getRollupDefinitionsToUse(rollupType, CRLP_RollupProcessingOptions.RollupTypeFilter.All));
+                if (!String.isEmpty(addlFilter)) {
+                    whereClauses.add(addlFilter);
+                }
+            }
+
+            // Exclude Organization Donations from Contact Hard Credit rollups (where Opp.Account.SYSTEM_IsIndividual=false)
+            // When the npo02__Always_Rollup_to_Primary_Contact__c is field is unchecked.
+            if (summaryObjectType == Contact.SObjectType && householdSettings.npo02__Always_Rollup_to_Primary_Contact__c == false) {
+                whereClauses.add('Account.npe01__SYSTEMIsIndividual__c = True');
+            }
+
+            // Force the details query to sort by Ascending CreatedDate so that the records are process
+            // in the order in which they were created in the database. Thus the 'first' is truly the first
+            // and the last is truly the last.
+            orderByFields.add('CreatedDate ASC');
+        }
+
+        if (!whereClauses.isEmpty()) {
+            query += ' WHERE ' + String.join(whereClauses, ' AND ');
+        }
+        if (customWhereClauses != null && !customWhereClauses.isEmpty()) {
```

Comment:
What is the purpose of the `customWhereClauses`? Can the query builder have both `whereClauses` and `customWhereClauses`?

Reply:
`customWhereClauses` was meant to allow the caller to pass a specific additional where clause into the builder that isn't automatically created by the logic in this method. It's not actually implemented anywhere at the moment though. This is a good example of where I started thinking ahead to future refactors, but decided to not take it that far quite yet.

Reply:
Let's remove code if it is not required at this point?

Reply:
That's fine if you're good with not digging too much further into the query refactoring at this point.